script/tasks/diamondSyncSigs.ts

```diff
@@ -0,0 +1,116 @@
+import { defineCommand, runMain } from 'citty'
+import {
+  Hex,
+  createPublicClient,
+  createWalletClient,
+  getContract,
+  http,
+  parseAbi,
+  type Chain,
+} from 'viem'
+import { ethers } from 'ethers6'
+import * as chains from 'viem/chains'
+import { privateKeyToAccount } from 'viem/accounts'
+
+export const chainNameMappings: Record<string, string> = {
+  zksync: 'zkSync',
+  polygonzkevm: 'polygonZkEvm',
+}
+
+const chainMap: Record<string, Chain> = {}
+for (const [k, v] of Object.entries(chains)) {
+  // @ts-ignore
+  chainMap[k] = v
+}
+
+const main = defineCommand({
+  meta: {
+    name: 'diamond-sync-sigs',
+    description: 'Sync approved function signatures',
+  },
+  args: {
+    network: {
+      type: 'string',
+      description: 'Network name',
+      required: true,
+    },
+    rpcUrl: {
+      type: 'string',
+      description: 'RPC URL',
+    },
+    privateKey: {
+      type: 'string',
+      description: 'Private key',
+      required: true,
+    },
+  },
+  async run({ args }) {
+    const { network, privateKey } = args
+
+    const chainName = chainNameMappings[network] || network
+    const chain: Chain = chainMap[chainName]
+
+    const deployedContracts = await import(
+      `../../deployments/${network.toLowerCase()}.json`
+    )
+
+    const rpcUrl = args.rpcUrl || chain.rpcUrls.default.http[0]
+
+    const publicClient = createPublicClient({
```

Comment:
a little more commenting would improve the readability of the file and allow concepts like "what is a public client" here  to be easier/faster understood